src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -0,0 +1,2526 @@
+                    TraceSessionClosedResponseFailure(e);
+                }
+            }
+
+            internal Message CreateCloseMessage(CancellationToken token)
+            {
+                RequestSecurityToken rst = new RequestSecurityToken(this.Settings.SecurityStandardsManager);
+                rst.RequestType = this.Settings.SecurityStandardsManager.TrustDriver.RequestTypeClose;
+                rst.CloseTarget = this.Settings.IssuedSecurityTokenParameters.CreateKeyIdentifierClause(this.currentSessionToken, SecurityTokenReferenceStyle.External);
+                rst.MakeReadOnly();
+                Message closeMessage = Message.CreateMessage(this.messageVersion, ActionHeader.Create(this.Settings.SecurityStandardsManager.SecureConversationDriver.CloseAction, this.messageVersion.Addressing), rst);
+                RequestReplyCorrelator.PrepareRequest(closeMessage);
+                if (this.LocalAddress != null)
+                {
+                    closeMessage.Headers.ReplyTo = this.LocalAddress;
+                }
+                else
+                {
+                    if (closeMessage.Version.Addressing == AddressingVersion.WSAddressing10)
+                    {
+                        closeMessage.Headers.ReplyTo = null;
+                    }
+                    else if (closeMessage.Version.Addressing == AddressingVersion.WSAddressingAugust2004)
+                    {
+                        closeMessage.Headers.ReplyTo = EndpointAddress.AnonymousAddress;
+                    }
+                    else
+                    {
+                        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                            new ProtocolException(SR.Format(SR.AddressingVersionNotSupported, closeMessage.Version.Addressing)));
+                    }
+                }
+                this.securityProtocol.SecureOutgoingMessageAsync(closeMessage, token);
+                closeMessage.Properties.AllowOutputBatching = false;
+                return closeMessage;
+            }
+
+            protected void SendClose(CancellationToken token)
+            {
+                try
+                {
+                    using (Message closeMessage = CreateCloseMessage(token))
+                    {
+                        SendMessage(null, closeMessage, token);
+                    }
+                    TraceSessionClosedSuccess();
+                }
+                catch (CommunicationException e)
+                {
+                    TraceSessionClosedFailure(e);
+                }
+                catch (TimeoutException e)
+                {
+                    TraceSessionClosedFailure(e);
+                }
+            }
+
+
+
+            protected void SendMessage(RequestContext requestContext, Message message, CancellationToken token)
+            {
+
+                if (requestContext != null)
+                {
+                    requestContext.ReplyAsync(message, token);
```

Comment:
You aren't awaiting the ReplyAsync call which means the CloseAsync call will execute before the ReplyAsync call has completed. You must await it, and make this method async.